app/controllers/course-admin/code-example-insights.ts

```diff
@@ -0,0 +1,70 @@
+import { tracked } from '@glimmer/tracking';
+import { next } from '@ember/runloop';
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import type RouterService from '@ember/routing/router-service';
+import { service } from '@ember/service';
+import type CommunityCourseStageSolutionModel from 'codecrafters-frontend/models/community-course-stage-solution';
+import type CourseStageModel from 'codecrafters-frontend/models/course-stage';
+import type LanguageModel from 'codecrafters-frontend/models/language';
+import type { CodeExampleInsightsRouteModel } from 'codecrafters-frontend/routes/course-admin/code-example-insights';
+import type AuthenticatorService from 'codecrafters-frontend/services/authenticator';
+
+export default class CodeExampleInsightsController extends Controller {
+  @service declare authenticator: AuthenticatorService;
+  @service declare router: RouterService;
+
+  @tracked expandedSolution: CommunityCourseStageSolutionModel | null = null;
+  queryParams = ['sort_mode'];
```

Comment:
Check what other files do -- queryParams appears either after all @tracked invocations or before all of them. I think it's before